pennylane/templates/subroutines/qsvt.py

```diff
@@ -455,9 +455,16 @@ def compute_matrix(*args, **kwargs):
 
 def _qsp_to_qsvt(angles):
     r"""Converts qsp angles to qsvt angles."""
+    num_angles = len(angles)
     new_angles = qml.math.array(copy.copy(angles))
-    new_angles[0] += 3 * np.pi / 4
-    new_angles[-1] -= np.pi / 4
+    indicies = qml.math.convert_like(np.arange(num_angles), angles)
 
-    new_angles[1:-1] += np.pi / 2
+    update_vals = np.zeros(num_angles)
+    update_vals[0] = 3 * np.pi / 4
+    update_vals[1:-1] = np.pi / 2
+    update_vals[-1] = -np.pi / 4
+
+    update_vals = qml.math.convert_like(update_vals, angles)
+
+    new_angles = qml.math.add_index(new_angles, indicies, update_vals)
```

Comment:
Wait I think in this case we just have:
```py
new_angles = angles + update_vals
```
no? 😅

Reply:
Nice catch, I definitely over engineered this feature. I removed the `add_index` call and used your suggestion. I kept the feature addition of `add_index()` is it might be useful tool to support jax jit compatible in place addition in the future.